### PR TITLE
[contract-proxy-kit] Add definition

### PR DESCRIPTION
**Pull request by cag**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
<s>- [ ] Test the change in your own code. (Compile and run.)</s>

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/contract-proxy-kit/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// Project: https://github.com/gnosis/contract-proxy-kit#readme
+// Definitions by: Alan Lu <https://github.com/cag>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.6
+
+import { ethers } from 'ethers';
+
+export = CPK;
+
+interface NetworkConfigEntry {
```
Comment (uniqueiniquity) [inline types/contract-proxy-kit/index.d.ts]: If you want your users to be able to import these types from the module, you should wrap them in a namespace called `CPK`.

File: types/contract-proxy-kit/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Type definitions for contract-proxy-kit 1.0
+// Project: https://github.com/gnosis/contract-proxy-kit#readme
+// Definitions by: Alan Lu <https://github.com/cag>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.6
```
Comment (uniqueiniquity) [inline types/contract-proxy-kit/index.d.ts]: Is this really the minimum version of TypeScript that is compatible with these type definitions?

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/contract-proxy-kit/index.d.ts (hunk 2 above)
Comment (cag) [reply types/contract-proxy-kit/index.d.ts]: Currently, yes, because accessors were introduced in 3.6. Should I instead be making it a readonly property?

Comment (uniqueiniquity) [reply types/contract-proxy-kit/index.d.ts]: No, that makes sense. Thanks!

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Review by uniqueiniquity — APPROVED** [review]

**Pull request merged**